exporter/clickhouseexporter/version_info.go

```diff
@@ -0,0 +1,39 @@
+package clickhouseexporter
+
+import (
+	"runtime"
+	"runtime/debug"
+	"strings"
+	"sync"
+)
+
+var (
+	once          sync.Once
+	cachedVersion string
+)
+
+func getCollectorVersion() string {
+	once.Do(func() {
+		osInformation := runtime.GOOS[:3] + "-" + runtime.GOARCH
+		unknownVersion := "unknown-" + osInformation
+
+		info, ok := debug.ReadBuildInfo()
+		if !ok {
+			cachedVersion = unknownVersion
+			return
+		}
+
+		for _, mod := range info.Deps {
```

Comment:
may be info.Main.version is the simpler way. ref: https://github.com/open-telemetry/opentelemetry-collector/blob/c4e527e7e8e099be7c21554d622bb489c6890da4/cmd/builder/internal/version.go#L20

Reply:
Much better, thank you for finding this!